wrapper/src/main/java/software/amazon/jdbc/HikariPooledConnectionProvider.java

```diff
@@ -98,6 +103,8 @@ public HikariPooledConnectionProvider(
       HikariPoolConfigurator hikariPoolConfigurator, HikariPoolMapping mapping) {
     this.poolConfigurator = hikariPoolConfigurator;
     this.poolMapping = mapping;
+    final LeastConnectionsHostSelector hostSelector = new LeastConnectionsHostSelector(databasePools);
+    acceptedStrategies.put(LeastConnectionsHostSelector.STRATEGY_LEAST_CONNECTIONS, hostSelector);
```

Comment:
You can't add this strategy to a static map. The strategy belongs to a particular instance of this connection provider and uses it resources (databasePools). If you add it to a static collection, other connection providers may use it and it's bad.